doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,185 @@
+
+This optimization is called _filter pushdown_... and it can save a substantial amount of time and computation given a large collection and a selective filter. This tends to be common for append-only timeseries datasets and temporal filters, but it can be useful for any dataset where most queries look at only a particular "range" of the data.
+
+Materialize will always try to filter out parts it doesn't need for a particular query, but that filtering is usually only effective when similar rows are stored together.
+If you want to make sure that filter pushdown reliably kicks in for your query, you can:
+
+- Use a `PARTITION BY` clause on the relevant column to ensure that data with similar values for that column are stored close together.
+- Add a filter to your query that only returns true for a narrow range of values in that column.
+
+Filters that consist of arithmetic, date math, and comparisons are generally eligible for pushdown, including all the examples in this page. However, more complex filters might not be. You can check whether the filters in your query can be pushed down using [an `EXPLAIN` statement](/sql/explain-plan/). In the following example, we can be confident our temporal filter will be pushed down because it's present in the `pushdown` list at the bottom of the output.
+
+```mzsql
+EXPLAIN SELECT * FROM events WHERE mz_now() <= event_ts + INTERVAL '5min';
+----
+Explained Query:
+[...]
+Source materialize.public.events
+  [...]
+  pushdown=((mz_now() <= timestamp_to_mz_timestamp((#0 + 00:05:00))))```
+```
+
+Some common functions, such as casting from a string to a timestamp, can prevent filter pushdown for a query. For similar functions that _do_ allow pushdown, see [the pushdown functions documentation](/sql/functions/pushdown/).
+
+## Examples
+
+These examples create real objects. After you have tried the examples, make sure to drop these objects and spin down any resources you may have created.
+
+### Partitioning by timestamp
+
+For timeseries or "event"-type collections, it's often useful to partition the data by timestamp.
+
+1. First, create a table called `events`.
+    ```mzsql
+    -- Create a table of timestamped events. Note that the `event_ts` column is
+    -- first in the column list and in the parition-by clause.
+    CREATE TABLE events (
+        event_ts timestamptz,
+        content text
+    ) WITH (
+        PARTITION BY (event_ts)
+    );
+    ```
+
+1. Insert a few records, one "older" record and one more recent.
+    ```mzsql
+    INSERT INTO events VALUES (now()::timestamp - '5 minutes', 'hello');
+    INSERT INTO events VALUES (now(), 'world');
+    ```
+
+1. Run a select statement against the data sometime in the next few minutes. This should return only the more recent of the two rows.
```

Comment:
We can remove the "sometime in the next few minutes" yes? It kind of makes it sound like people should wait ... as opposed to don't wait too long before running this ... but I think people should be able to tell.

We could explicitly state:
Run the following select statement to return events that occurred within the last two minutes.  This should return ...

Reply:
"Don't wait too long before running this" is definitely the point I want to emphasize. (Because if you wait too long the result changes.)

A rewrite like:

> Run the following select statement to return events that occurred within the last two minutes. This should return only the more recent of the two rows.

Would be incorrect: you need to say something about when the query is being run to say what the results will be.

Reply:
Yeah, they probably will! But I'd rather be precise now so I don't get a bug report later. 😅

I've switched to "within the next five minutes". Language like "immediately" would also be fine I think!